test/NuGet.Core.Tests/NuGet.Commands.Test/BuildAssetsUtilsTests.cs

```diff
@@ -843,5 +841,63 @@ public async Task BuildAssetsUtils_GeneratePathPropertyForTools(bool hasTools)
                 }
             }
         }
+
+        [Fact]
+        public void BuildAssetsUtils_GenerateProjectRelativeAssetsFilePath()
+        {
+            using (var workingDir = TestDirectory.Create())
+            {
+                // Arrange
+                var filePath = Path.Combine(workingDir, "obj", "test.props");
+                var assetsPath = Path.Combine(workingDir, "obj", "project.assets.json");
+
+                var doc = BuildAssetsUtils.GenerateEmptyImportsFile();
+                var file = new MSBuildOutputFile(filePath, doc);
+
+                // Act
+                BuildAssetsUtils.AddNuGetPropertiesToFirstImport(
+                    new[] { file },
+                    Enumerable.Empty<string>(),
+                    string.Empty,
+                    ProjectStyle.PackageReference,
+                    assetsPath,
+                    success: true);
+
+                var props = TargetsUtility.GetMSBuildProperties(doc);
+
+                // Assert
+                Assert.Equal("$(MSBuildThisFileDirectory)project.assets.json", props["ProjectAssetsFile"]);
+            }
+        }
+        // TODO NK - Add better tests, can the assets file and the props file ever be different?
+        // Would this work for project.json?
```

Comment:
Address feedback before merging and remove comment.

Reply:
Addressed. 
I really need to do a better job at removing those comments 😞